xbmc/GUILargeTextureManager.cpp

```diff
@@ -121,11 +124,15 @@ bool CGUILargeTextureManager::CLargeTexture::DeleteIfRequired(bool deleteImmedia
   return false;
 }
 
-void CGUILargeTextureManager::CLargeTexture::SetTexture(CTexture* texture)
+void CGUILargeTextureManager::CLargeTexture::SetTexture(std::unique_ptr<CTexture> texture)
 {
   assert(!m_texture.size());
   if (texture)
-    m_texture.Set(texture, texture->GetWidth(), texture->GetHeight());
+  {
+    const auto width = texture->GetWidth();
+    const auto height = texture->GetHeight();
+    m_texture.Set(std::move(texture), width, height);
+  }
```

Comment:
is this done for readability?
`m_texture.Set(std::move(texture), texture->GetWidth(), texture->GetHeight());`
seems to work fine (on linux at least) even though texture is moved.

Reply:
ok, answering to myself: The order of evaluation of arguments is unspecified. i‘m going to merge this tomorrow